web/src/reactions.ts

```diff
@@ -460,55 +528,96 @@ export function set_clean_reactions(message) {
     // to make sure reify_message_id moves the data structure
     // properly.
     message.clean_reactions = new Map();
+
+    const reaction_counts_and_user_ids = [...distinct_reactions.keys()].map((local_id) => {
```

Comment:
The other main change I was working on is getting the `clean_reaction` built all at once instead of half-building it, adding it to the message, and then passing it to a function to complete building it, since that was really annoying to type.

Does this seem like the right way to go about it? It feels a bit messy still.